src/Integration/MefServices/ActiveSolutionBoundTracker.cs

```diff
@@ -102,17 +108,28 @@ public ActiveSolutionBoundTracker(IHost host, IActiveSolutionTracker activeSolut
 
             if (isSolutionCurrentlyBound)
             {
-                var solutionBinding = this.extensionHost.GetService<ISolutionBindingSerializer>();
-                solutionBinding.AssertLocalServiceIsNotNull();
-                BoundSonarQubeProject boundProject = solutionBinding.ReadSolutionBinding();
-                var connectionInformation = boundProject.CreateConnectionInformation();
-
-                // TODO: handle connection failure
-                // TODO: block until the connection is complete
-                await this.extensionHost.SonarQubeService.ConnectAsync(connectionInformation, CancellationToken.None);
+                try
+                {
+                    var connectionInformation = GetConnectionInformation();
+                    await sonarQubeService.ConnectAsync(connectionInformation, CancellationToken.None);
+                }
+                catch (Exception ex)
```

Comment:
Don't know how useful it could be but isn't it better to create some helper that will handle the different catch as in `ConnectionWorkflow.cs` and display different messages?

Reply:
Makes sense.